### ISSUE TITLE
ComposeViewModelForwarding produces false negative results

### ISSUE THREAD

**Issue by Ilya-Gh** [slackhq/compose-lints#148]
The following codes report the error:

```kotlin
@Composable
fun ComposableOk(
    viewModel: ViewModel,
    modifier: Modifier = Modifier
) {
    ComposableWithViewModel(viewModel = viewModel)
}
```

In the following screnario, the rule fails to detect the issue when the viewModel is transferred between Composable layers, leading to a false-negative:

```kotlin
@Composable
fun ComposableFalseNegative(
    viewModel: ViewModel,
    modifier: Modifier = Modifier
) {
    Row {
        ComposableWithViewModel(viewModel = viewModel)
    }
}
```

**Comment by ZacSweers** [comment]
Can you please paste the error or screenshot?

**Comment by Ilya-Gh** [comment]
@ZacSweers Sorry for not stating it clear enough. The issue is that the ComposeViewModelForwarding rule does not report the error when it should.

The case:

```kotlin
@Composable
fun ComposableFalseNegative(
    viewModel: ViewModel,
    modifier: Modifier = Modifier
) {
    Row {
        ComposableWithViewModel(viewModel = viewModel)
    }
}
```

**Comment by mkaomwakuni** [comment]
Could be the ViewModel  logic since this approach makes it hard to reuse or replace it  some cases
